### PR TITLE
CI: Enable integration tests on Windows

### PR DESCRIPTION
**Pull request by mayankshah1607**
Signed-off-by: Mayank Shah <mayankshah1614@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by mayankshah1607** [comment]
Not really sure why we're getting the following error. Seemed to work on my local machine
```
disk_test.go:26: 
        	Error Trace:	disk_test.go:26
        	Error:      	Expected nil, but got: &status.statusError{state:impl.MessageState{NoUnkeyedLiterals:pragma.NoUnkeyedLiterals{}, DoNotCompare:pragma.DoNotCompare{}, DoNotCopy:pragma.DoNotCopy{}, atomicMessageInfo:(*impl.MessageInfo)(nil)}, sizeCache:0, unknownFields:[]uint8(nil), Code:2, Message:"Could not get page83 ID: IOCTL_STORAGE_QUERY_PROPERTY failed: Incorrect function.", Details:[]*anypb.Any(nil)}
```

**Comment by mayankshah1607** [comment]
@ddebroy @andyzhangx Have the following operations worked before - https://github.com/kubernetes-csi/csi-proxy/blob/master/integrationtests/filesystem_test.go#L66

On my local machine as well as on the CI, it is giving errors creating symlink due to permissions. I have tried to run the proxy as well as the tests as admin. Seems related to - https://github.com/golang/dep/issues/773

**Comment by andyzhangx** [comment]
> @ddebroy @andyzhangx Have the following operations worked before - https://github.com/kubernetes-csi/csi-proxy/blob/master/integrationtests/filesystem_test.go#L66
> 
> On my local machine as well as on the CI, it is giving errors creating symlink due to permissions. I have tried to run the proxy as well as the tests as admin. Seems related to - [golang/dep#773](https://github.com/golang/dep/issues/773)

@mayankshah1607 No, those tests have never been run on Windows, so if it does not work now, just skipIfRunOnWindows, add a todo comment to investigate later, thanks.
Let's add a basic unit test CI on Windows first!

**Comment by mayankshah1607** [comment]
@andyzhangx I have added separate job for unit test and integration test. Looks like some of the integration tests aren't running, so I have skipped them for now. PTAL, thanks!

File: integrationtests/csi_api_gen_test.go [hunk 1]
```diff
@@ -18,15 +18,18 @@ import (
...
 	// clean slate
 	require.Nil(t, os.RemoveAll("csiapigen/new_group/actual_output"))
-
 	logLevel := "3"
 	stdout, _ := runGenerator(t, "TestNewAPIGroup",
 		"--input-dirs", "github.com/kubernetes-csi/csi-proxy/integrationtests/csiapigen/new_group/api",
 		// might as well check that logging CLI args work as expected
 		"-v", logLevel)
-
+	t.Log("!!")
```
Comment (andyzhangx) [inline integrationtests/csi_api_gen_test.go:32]: remove this

File: integrationtests/csi_api_gen_test.go [hunk 2]
```diff
@@ -18,15 +18,18 @@ import (
 
 // TestNewAPIGroup tests that bootstraping a new group works as intended.
 func TestNewAPIGroup(t *testing.T) {
+	if os.Getenv("CSI_PROXY_GH_ACTIONS") == "TRUE" {
```
Comment (andyzhangx) [inline integrationtests/csi_api_gen_test.go:21]: why not use `isRunningOnGhActions` ?

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**